### ISSUE TITLE
Performance Degradation

### ISSUE THREAD

**Issue by pkrogel-kws** [Nexus-Mods/Vortex#143]
Seemingly without cause, it is now taking me around ~5 minutes for Vortex to deploy mods where as it used to take around 1.  Actually, I think this is less of an issue with the actual deployment vs the sorting and whatever else it does prior, as it also takes around this long to finish processing when I update mod rules.  

I've tried rolling back to 0.15.0 as well as 0.14.3 with no change in performance.  MO2 also seems to work just fine, as I tried that to rule out anything file-system related on my end.

Any ideas to help restore normal performance?  Is there anything I can do to allocate more resources?  Is there a way to clean up mod rules? I know I've seen lingering rules for mods that have since been uninstalled and removed.  I also know that I have a bunch bunch of "duplicate" rules that stem back from some issues I was having with Vortex before I disabled Microsoft One Drive (ex a rule on Mod A to load before Mob B, and then an additional rule on Mob B to load before Mod A.)  

Some additional info:
Game - SkyrimVR
Mods - 320+ total, ~150 enabled
Vortex 0.15.1
Windows 10 Home x64 (17134.48)
i7-6700HQ
16GB DDR4
GTX-1080 Ti
Crucial MX300 1TB CT1050MX300SSD1
---

On a unrelated note -- what is your opinion on Electron at this point?  I've followed React since it's inception, and have since migrated all of our organization's web applications over with great success.  We have a couple potential projects coming down the pipe that would be "desktop" applications (of which I have 0 experience).  Obviously, the thought of leveraging React to share code and reuse existing existing application logic seems pretty attractive, but I've heard mixed opinions on Electron and it's performance.

**Comment by TanninOne** [comment]
The performance issue has already been fixed in commit e55e8df840839cb6744ce2c9e2764a92828d7ebf, a new release is imminent.

Regarding electron and react: I like the idea behind both, they are clearly how things should work in the future and I wouldn't worry about performance too much. It takes more work to get an electron application to perform well but it can be done and if you come from a web background anyway, you won't know what you're missing out with stuff like Qt anyway.

I do feel that they are not polished enough. Tbh, considering all the buzz they are getting, I was expecting far more sophisticated software when I went into this project. I'm not suuuper convinced of the qualification of developers in either project actually. I have reported issues to both and responses have sometimes been disheartening.
E.g. I reported a sequence of calls at startup that would crash electron and there was no documentation that you shouldn't do that, the issue was closed almost immediately with the "argument" it was undefined behaviour. As if undefined behaviour due to laziness was acceptable.
On react I reported an issue that they relied on the node cache mechanism for a singleton implementation despite the fact node clearly says you can't do that and despite the fact I could demonstrate a situation on windows where it in fact didn't work. Closed, because the alternative solution "didn't feel right" to him.

Another big issue is: Desktop UIs tend to be much more complex than web UIs, the ui frameworks aren't really up to the task yet.
For example with Qt if you try to display a table of data, stuff like resizeable columns, rearrangeable columns, toggling columns, sorting, filtering, optimised rendering of only the visible area, hierarchical data, drag and drop reordering, ... all comes standard with the default control, you just check some checkboxes which features you want/how you want them to behave and throw the data at it.
html tables can't do any of what I listed out of the box. There are tons of table libraries on the net, most are abandoned, barely two have the same way of providing the data, none has even half those features and I've yet to find a table library where I didn't find a bug within 2 minutes.
tl;dr: quality of web frameworks/libraries is hilariously bad compared to desktop frameworks like qt, gtk, wxwidgets and since the js community tends to rewrite instead of improving I don't see that changing any time soon. The result is that you have to do a lot more work yourself, some things that took 5 minutes with Qt now take days.
Other things go quicker of course but it can still be frustrating when you know how easy it could be.

If you don't require cross-platform support, a fancy, highly customised UI or to share code between website and desktop application you will probably get results in less time from a desktop toolkit.
